Remove ActiveIssue for S.S.C.Xml tests that rely on DSA key generation

vcsjones opened the pull request:
MacOS cannot generate DSA keys, so make a pre-generated key and use that.

Closes #20575

vcsjones committed:
Remove ActiveIssue for S.S.C.Xml tests that rely on DSA key generation.

MacOS cannot generate DSA keys, so make a pre-generated key and use that.

vcsjones commented on src/libraries/System.Security.Cryptography.Xml/tests/DSAKeyValueTest.cs:
```diff
@@ -51,39 +51,42 @@ public void Ctor_Dsa_Null()
-            //Existing elements MUST be convertible from BASE64
-            elements.Select(element => Convert.FromBase64String(element.InnerText));
+                //Existing elements MUST be convertible from BASE64
+                elements.Select(element => Convert.FromBase64String(element.InnerText));
```
Heh. This doesn't do anything I think because Linq lazy evaluates. I'll fix this up.

bartonjs commented:
Me: But this feels like we're losing coverage from keys other than this one specific key...

AlsoMe: That's technically true, but there's not a whole lot in these tests that cares.

Me: What if there was something that failed in `AsymmetricDSASignature` because of some heretofore unseen key characteristic, akin to an RSA key having a diminished DP value?

AlsoMe: Then someone'd probably just hit retry and we'd never notice it.

Me: It'd still turn up in a report somewhere?

AlsoMe: The code is 17 years old, and has barely ever changed.  These days, you don't even really like the type existing, much less taking changes in it.  Also, it's DSA.  Does anyone even use DSA? If so, they're... (how should I say it?) ... wrong for doing so.  There's nothing left to test.  It'd have either been found by now, or won't ever be.

Me: Fair enough.

bartonjs commented:
Doesn't compile on NetFX :(

vcsjones commented:
> Doesn't compile on NetFX :(

_1000-yard stare_. Right. This needs to compile for .NET Framework.

vcsjones committed:
Fix assert to actually do something meaningful

vcsjones committed:
Fix compilation for .NET Framework.

Only used the fixed key on macOS so other platforms continue to test with a random key.

vcsjones commented:
> Me: But this feels like we're losing coverage from keys other than this one specific key.

Kevin: Eh let's just condition it on macOS.
Other Kevin: He said it was fi—
Kevin: It's like 5 lines of code. We're doing it.

vcsjones committed:
Use a random key where possible
